src/buildstream/sandbox/_sandboxbuildboxrun.py

```diff
@@ -32,6 +34,27 @@
 # BuildBox-based sandbox implementation.
 #
 class SandboxBuildBoxRun(SandboxREAPI):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        context = self._get_context()
+        cascache = context.get_cascache()
+
+        re_specs = context.remote_execution_specs
+        if re_specs and re_specs.action_spec and not re_specs.exec_spec:
```

Comment:
Is there a value in excluding this when we have `re_specs.exec_spec`? IIUC, we can use this sandbox even though remote-execution is configured in code paths that go through `Element._prepare_sandbox()` (which sets allow_remote=False) such as `bst shell`.

Is there a reason to prevent using the action cache in that situation?

Reply:
When we add action cache support for BuildStream actions, we need to make sure that we're not attempting to cache interactive shells. However, I don't think we need to disable remote execution or action cache access for sandboxed REAPI clients, so I've removed this condition.